js/compressed-token/src/program.ts

```diff
@@ -0,0 +1,241 @@
+        mockConnection,
+        useWallet(mockKeypair),
+        confirmConfig,
+      );
+      setProvider(mockProvider);
+      this._program = new Program(IDL, this.programId, mockProvider);
+    }
+  }
+
+  static deriveMintAuthorityPda = (
+    authority: PublicKey,
+    mint: PublicKey,
+  ): PublicKey => {
+    const [pubkey] = PublicKey.findProgramAddressSync(
+      [
+        MINT_AUTHORITY_SEED, // eq Buffer.from
+        authority.toBuffer(),
+        mint.toBuffer(),
+      ],
+      this.programId,
+    );
+    return pubkey;
+  };
+
+  static deriveTokenPoolPda(mint: PublicKey): PublicKey {
+    const seeds = [POOL_SEED, mint.toBuffer()];
+    const [address, _] = PublicKey.findProgramAddressSync(
+      seeds,
+      this.programId,
+    );
+    return address;
+  }
+
+  static get cpiAuthorityPda(): PublicKey {
+    const [address, _] = PublicKey.findProgramAddressSync(
+      [Buffer.from('cpi_authority')],
+      this.programId,
+    );
+    return address;
+  }
+
+  static async createMint(
+    params: CreateMintParams,
+  ): Promise<TransactionInstruction[]> {
+    const mint = params.mint;
+    const authority = params.authority;
+    const feePayer = params.feePayer;
+    const rentExemptBalance = params.rentExemptBalance;
+
+    const createMintAccountInstruction = SystemProgram.createAccount({
+      fromPubkey: feePayer,
+      lamports: rentExemptBalance,
+      newAccountPubkey: mint,
+      programId: TOKEN_PROGRAM_ID,
+      space: MINT_SIZE,
+    });
+
+    const mintAuthorityPda = this.deriveMintAuthorityPda(authority, mint);
+
+    const initializeMintInstruction = createInitializeMint2Instruction(
+      mint,
+      params.decimals,
+      mintAuthorityPda,
+      params.freezeAuthority,
+      TOKEN_PROGRAM_ID,
+    );
+
+    const fundAuthorityPdaInstruction = SystemProgram.transfer({
+      fromPubkey: feePayer,
+      toPubkey: mintAuthorityPda,
+      lamports: rentExemptBalance, // TODO: check that this is the right PDA size
+    });
+
+    const tokenPoolPda = this.deriveTokenPoolPda(mint);
+
+    const ix = await this.program.methods
+      .createMint()
+      .accounts({
+        mint,
+        feePayer,
+        authority,
+        tokenPoolPda,
+        systemProgram: SystemProgram.programId,
+        mintAuthorityPda,
+        tokenProgram: TOKEN_PROGRAM_ID,
+      })
+      .instruction();
+
+    return [
+      createMintAccountInstruction,
+      initializeMintInstruction,
+      fundAuthorityPdaInstruction,
+      ix,
+    ];
+  }
+
+  static async mintTo(params: MintToParams): Promise<TransactionInstruction> {
+    const systemKeys = defaultStaticAccountsStruct();
+
+    const { mint, feePayer, authority, merkleTree, toPubkey, amount } = params;
+
+    const tokenPoolPda = this.deriveTokenPoolPda(mint);
+    const mintAuthorityPda = this.deriveMintAuthorityPda(authority, mint);
+
+    const amounts = toArray(amount).map((amount) => bn(amount));
```

Comment:
why do you need to do toArray here?

Reply:
yup it's typesafe, it can either be one amount or an array of amounts in case you're batch-minting.
we're allowing non-arrays for  convenience when you mint a single instance